frontend/src/components/Navbar/NavBar.tsx

```diff
@@ -48,9 +61,23 @@ const Navbar: React.FC<NavbarProps> = ({ setIsCollapsed }) => {
         setShowFeedbackRatingPanel(false);
         setSettingsPanel(false);
     }
+
+
+    useEffect(() => {
```

Comment:
@egdagger, please consider moving this code to the context provider located at frontend/src/providers/AppProviders.tsx. By doing so, it will be included in the global state, allowing the entire application to access whether the current user has an active Financial Assistant status.